### ISSUE TITLE
v2.7.0 and higher is missing 'policy.qouta.sync.table.connection'

### ISSUE THREAD

**Issue by MickeSpike** [Azure/api-management-self-hosted-gateway#316]
### Report

Hello.

In my organization we use the self hosted gateway on premise.
We have discovered that in later versions Self-hosted Gateway a setting (**policy.qouta.sync.table.connection**) is missing in appsettings.json
This causes the logs to contain a lot warnings, a new one every 30th second, about not being able to find the setting.

```
Warn] 2024-11-26T11:42:28.440 [QuotaCounterSyncFailedUnexpectedly], message: Unexpected fail occured while performing synchronization, exception: System.ArgumentException: Failed to initialize storage table client using setting 'policy.qouta.sync.table.connection'
 ---> System.Configuration.ConfigurationErrorsException: Setting 'policy.qouta.sync.table.connection' could not be found.
	at Microsoft.Azure.ApiManagement.Infra.Configuration.SettingsProviderExtensions.GetSetting(ISettingsProvider provider, String name)
	at Gateway.Configuration.Storage.Clients.StorageClientBuilder.BuildTableClientCore(String connectionStringSetting, SettingWithFallback`1 tableNameSettingWithFallback, Boolean deploymentSpecific, Nullable`1 clientTimeoutSetting, Nullable`1 clientRetrySetting, TokenCredential tokenCredential, Boolean useTunnelProxy, Func`2 transformConnectionStrings)
	at Gateway.Configuration.Storage.Clients.StorageClientBuilder.BuildTableClient(String connectionStringSetting, SettingWithFallback`1 tableNameSettingWithFallback, TokenCredential tokenCredential, Boolean deploymentSpecific, Nullable`1 clientTimeoutSetting, Nullable`1 clientRetrySetting, Boolean useTunnelProxy, Func`2 transformConnectionStrings)
   --- End of inner exception stack trace ---
	at Gateway.Configuration.Storage.Clients.StorageClientBuilder.BuildTableClient(String connectionStringSetting, SettingWithFallback`1 tableNameSettingWithFallback, TokenCredential tokenCredential, Boolean deploymentSpecific, Nullable`1 clientTimeoutSetting, Nullable`1 clientRetrySetting, Boolean useTunnelProxy, Func`2 transformConnectionStrings)
	at Gateway.Host.AspNetCore.ExternalConfiguration.BackwardsCompatibility.LazyTransactionalTable.<Initialize>g__CreateTable|10_0(String connectionString)
	at Gateway.Configuration.Client.ConfigurationServiceTableStorageWrapper`1.TryGetTable(ITransactionalTable`1& result)
	at Gateway.Configuration.Client.ConfigurationServiceTableStorageWrapper`1.ExecuteAsync[TResult](Func`3 action, CancellationToken cancellation)
	at Gateway.Host.AspNetCore.ExternalConfiguration.BackwardsCompatibility.LazyTransactionalTable.SaveAsync(String partitionKey, IEnumerable`1 entities, String transactionId, CancellationToken cancellationToken)
	at Microsoft.WindowsAzure.ApiManagement.Proxy.Api.Controllers.CountersStorageController.SynchronizeAsync(IEnumerable`1 counters, String batchId, CancellationToken cancellationToken)
	at Microsoft.WindowsAzure.ApiManagement.Proxy.Api.Controllers.CountersStorageController.ExchangeCounterStatesAsync(SmartCountersUpdateSet updateSet, CancellationToken cancellationToken)
	at Gateway.Quota.Services.QuotaByKeyCounterService.ExchangeSynchronizationDataAsync(List`1 countersChanges, CancellationToken cancellationToken)
	at Gateway.Quota.Services.QuotaByKeyCounterService.PerformSynchronization(CancellationToken cancellationToken)
	at Gateway.Quota.Services.QuotaByKeyCounterService.Synchronize(CancellationToken cancellationToken)
```
This is seen in v 2.7.0 and higher. Eg:
```
mcr.microsoft.com/azure-api-management/gateway:2.7.0
mcr.microsoft.com/azure-api-management/gateway:latest
```
However, lower versions seems to be ok. Eg:
```
mcr.microsoft.com/azure-api-management/gateway:2.6.0
```

### Expected Behavior

The warnings about the missing settings, 'policy.qouta.sync.table.connection', should be removed.
Either by handling in code or by adding it to the appsettings.json as in the curernt state it floods the logs.

### Actual Behavior

Self-hosted gateway contains a lot of warnings.

### Steps to Reproduce the Problem

1. Create a new self-hosted gateway in the portal.
2. Start a container using the default settings and configuration. Ex:
```
    podman run -d -p 80:8080 -p 443:8081 --name demo-gateway --env-file env.conf mcr.microsoft.com/azure-api-management/gateway:latest
```
3. Run a request against the gateway. 
4. Check container logs.

### Logs from self-hosted gateway

```
Warn] 2024-11-26T11:42:28.440 [QuotaCounterSyncFailedUnexpectedly], message: Unexpected fail occured while performing synchronization, exception: System.ArgumentException: Failed to initialize storage table client using setting 'policy.qouta.sync.table.connection'
 ---> System.Configuration.ConfigurationErrorsException: Setting 'policy.qouta.sync.table.connection' could not be found.
	at Microsoft.Azure.ApiManagement.Infra.Configuration.SettingsProviderExtensions.GetSetting(ISettingsProvider provider, String name)
	at Gateway.Configuration.Storage.Clients.StorageClientBuilder.BuildTableClientCore(String connectionStringSetting, SettingWithFallback`1 tableNameSettingWithFallback, Boolean deploymentSpecific, Nullable`1 clientTimeoutSetting, Nullable`1 clientRetrySetting, TokenCredential tokenCredential, Boolean useTunnelProxy, Func`2 transformConnectionStrings)
	at Gateway.Configuration.Storage.Clients.StorageClientBuilder.BuildTableClient(String connectionStringSetting, SettingWithFallback`1 tableNameSettingWithFallback, TokenCredential tokenCredential, Boolean deploymentSpecific, Nullable`1 clientTimeoutSetting, Nullable`1 clientRetrySetting, Boolean useTunnelProxy, Func`2 transformConnectionStrings)
   --- End of inner exception stack trace ---
	at Gateway.Configuration.Storage.Clients.StorageClientBuilder.BuildTableClient(String connectionStringSetting, SettingWithFallback`1 tableNameSettingWithFallback, TokenCredential tokenCredential, Boolean deploymentSpecific, Nullable`1 clientTimeoutSetting, Nullable`1 clientRetrySetting, Boolean useTunnelProxy, Func`2 transformConnectionStrings)
	at Gateway.Host.AspNetCore.ExternalConfiguration.BackwardsCompatibility.LazyTransactionalTable.<Initialize>g__CreateTable|10_0(String connectionString)
	at Gateway.Configuration.Client.ConfigurationServiceTableStorageWrapper`1.TryGetTable(ITransactionalTable`1& result)
	at Gateway.Configuration.Client.ConfigurationServiceTableStorageWrapper`1.ExecuteAsync[TResult](Func`3 action, CancellationToken cancellation)
	at Gateway.Host.AspNetCore.ExternalConfiguration.BackwardsCompatibility.LazyTransactionalTable.SaveAsync(String partitionKey, IEnumerable`1 entities, String transactionId, CancellationToken cancellationToken)
	at Microsoft.WindowsAzure.ApiManagement.Proxy.Api.Controllers.CountersStorageController.SynchronizeAsync(IEnumerable`1 counters, String batchId, CancellationToken cancellationToken)
	at Microsoft.WindowsAzure.ApiManagement.Proxy.Api.Controllers.CountersStorageController.ExchangeCounterStatesAsync(SmartCountersUpdateSet updateSet, CancellationToken cancellationToken)
	at Gateway.Quota.Services.QuotaByKeyCounterService.ExchangeSynchronizationDataAsync(List`1 countersChanges, CancellationToken cancellationToken)
	at Gateway.Quota.Services.QuotaByKeyCounterService.PerformSynchronization(CancellationToken cancellationToken)
	at Gateway.Quota.Services.QuotaByKeyCounterService.Synchronize(CancellationToken cancellationToken)
```

### Self-hosted Gateway Version

>= 2.7.0

### Deployment

Docker (Compose)

### Platform

Other

### Kubernetes Version

None

### Anything else?

Also, the setting is mispelt. qouta vs. quota (policy.**qouta**.sync.table.connection). 
This is also true for many other settings in appsettings.json (even if it does not affect functionaility). Ex:

```
"policy.**qouta**.sync.interval": 
"policy.**qouta**.sync.operation.timeout":
"policy.**qouta**.notification.threshold.calls":
"policy.**qouta**.notification.threshold.bandwidth":
...
```

**Comment by tomkerkhove** [comment]
This is a bug that has been fixed and will be in 2.8.0 which is ETA before EOY

**Comment by MickeSpike** [comment]
Great! Excellent news.
Thank you very much for the quick reply.
